addon/instance-initializers/prefetch.js

```diff
@@ -38,7 +38,13 @@ export function initialize(instance) {
       }
 
       // Run the prefetch hook if the route has one.
-      handlerInfo.handler._prefetched = handlerInfo.runSharedModelHook(transition, 'prefetch', [fullParams]);
+      if (!handlerInfo.handler && handlerInfo.handlerPromise) {
+        handlerInfo.handlerPromise.then((handler) => {
+          handler._prefetched = handlerInfo.runSharedModelHook(transition, 'prefetch', [fullParams]);
+        });
+      } else {
+        handlerInfo.handler._prefetched = handlerInfo.runSharedModelHook(transition, 'prefetch', [fullParams]);
```

Comment:
Good point. I was only thinking about support for the new version, but this is more or less duck-typing for the version of Ember, which makes sense. Thanks!